lib/image-stream/archive.js

```diff
@@ -81,8 +80,20 @@ const extractEntryByPath = (archive, filePath, options) => {
     return Bluebird.resolve(options.default);
   }
 
-  return options.hooks.extractFile(archive, options.entries, fileEntry.name)
-    .then(rindle.extract);
+  return options.hooks.extractFile(archive, options.entries, fileEntry.name).then((stream) => {
+    return new Bluebird((resolve, reject) => {
+      let data = '';
```

Comment:
Would the pre-allocated buffer work if the uncompressed file size is larger than the free memory?

In fact, does the current code work if the file being extracted is larger than the computer's free memory?